Updated config-libvirt role bringing inline with better ansible practices

kenhitchcock opened the pull request:
### What does this PR do?
PR is an improvement on the old config-libvirt role. The previous role was failing when i tried using it. 

### How should this be tested?
Automated tests are preferred, but not always doable - especially for infrastructure. Include commands to run your new feature, and also post-run commands to validate that it worked. (please use code blocks to format code samples)

Role can be included with a simple playbook as per the example in the README file. The role will install the basic packages and start the libvirt environment.

### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.
resolves #<number>

No issues open that i could find.

### Other Relevant info, PRs, etc.
Please provide link to other PRs that may be related (blocking, resolves, etc. etc.)

None

### People to notify
cc: @redhat-cop/infra-ansible

@redhat-cop/infra-ansible

ostefek99 commented:
You might want to include, maybe as an optional, `bridge-utils` in the list of packages to install. Might be useful to have it there since creating bridged interfaces is quite common.

kenhitchcock commented:
@ostefek99 thanks for the tip, I have included this now in the pull request.

@oybed @sabre1041 please let me know if you are happy with this role. 

Much appreciated.
Ken Hitchcock

kenhitchcock commented:
@oybed 

Thank you for the comments, it has brought my attention to some oversights definitely. I have now corrected the boolean value issues and changed the include to import. You were 100% correct, I have not been thinking clearly when I made that change. This is why we have code review :)

With regard to why I change the group install to individual packages, was to simplify the install to the required packages only if the user only wants to use the command line. The option to install virt-manager is managed by the boolean variable for it.

The following link shows the install requirements.
https://access.redhat.com/articles/1344173#Q_how-install-virtualization-packages

The reason the original role failed to work was that the group used was incorrect. It should have been "Virtualization Host" (unless I'm missing something obvious) not "@^virtualization-host-environment". I am more than happy to raise an issue for it if you still require it.

Finally, the old file (libvirt.yml) was left for reference but can be removed. Maybe I'm wrong in my configuration of the main.yml, but I find it far cleaner to use it to reference sub-tasks. You then have the added ability to wrap the tasks with different options/blocks etc.. This is just my way of doing things. Happy to be corrected, this is how we learn.

Again, apologies for the silly mistakes and I hope you will still consider my PR.

Thanks
Ken Hitchcock

oybed commented:
FYI - lint checks have been enabled on this repo. Please resolve conflicts and push the latest.

kenhitchcock commented:
@oybed 
Another liniting blocker. 

With this pull request its complaining about a file that's been removed/renamed. With this kind of test how do you remove files no longer required?

Again this might just be me not understanding, i will do some reading shortly to educate myself.

oybed commented:
Same comment as the other PR - just rebase your fork/branch (and push the updated version) and you should be good.

kenhitchcock commented:
@oybed 

Linting issues now resolved.

tylerauerbeck commented:
@kenhitchcock Can you rebase this and then we can get this merged.

kenhitchcock commented:
@tylerauerbeck rebased and ready for your review.

oybed reviewed (approved):
LGTM - will merge and submit a follow-up PR with some changes
